src/utils/message.js

```diff
@@ -20,6 +20,15 @@ export const shouldBeMuted = (
     return false; // never hide a message when narrowed to topic
   }
 
+  // This logic isn't quite right - we want to make sure we never hide a
+  // message that has a mention, even if we aren't in the "Mentioned" narrow.
+  // However, it's more complex to do that, and this code fixes the largest
+  // problem we have with muted mentioned messages, which is that they show up
+  // in the count for the "Mentions" tab, but not in the actual narrow.
```

Comment:
One confusing thing in this comment: it describes the state before this bit of code was added, but at the time the comment is added, it's no longer accurate. Muted mentioned messages do now show up in the actual narrow under the "Mentions" tab.

I didn't notice that when first reading this comment -- it's subtle when reading now, when we're thinking about this change itself :wink: and so the previous state is salient in our minds. But in a future when someone (us included) is just trying to understand how the code is and doesn't remember this bugfix, it'll be confusing because the comment is saying one thing and then the code is ensuring the opposite.

About to merge, so I'll just fix this.